certification/internal/engine/engine.go

```diff
@@ -279,3 +298,162 @@ func untar(dst string, r io.Reader) error {
 		}
 	}
 }
+
+func writeCertImage(img v1.Image) error {
+	config, err := img.ConfigFile()
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+	}
+
+	manifest, err := img.Manifest()
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+	}
+
+	digest, err := img.Digest()
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+	}
+
+	layers, err := img.Layers()
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+	}
+
+	rawConfig, err := img.RawConfigFile()
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+	}
+
+	size, err := img.Size()
+	if err != nil {
+		return fmt.Errorf("%w: %s", errors.ErrImageInspectFailed, err)
+	}
+
+	labels := convertLabels(config.Config.Labels)
+	layerSize := make([]pyxis.Layer, 10)
+
+	certImage := pyxis.CertImage{
+		DockerImageDigest: digest.String(),
+		DockerImageID:     manifest.Config.Digest.String(),
+		ImageID:           digest.String(),
+		ParsedData: &pyxis.ParsedData{
+			Architecture:  config.Architecture,
+			Command:       strings.Join(config.Config.Cmd, " "),
+			Created:       config.Created.String(),
+			DockerVersion: config.DockerVersion,
+			ImageID:       digest.String(),
+			Labels:        labels,
+			OS:            config.OS,
+			Size:          size,
+			//TODO: figure out how to get/convert this data from crane
+			UncompressedLayerSizes: layerSize,
+		},
+		RawConfig:         string(rawConfig),
+		SumLayerSizeBytes: sumLayerSizeBytes(layers),
+		//TODO: figure out how to get/convert this data from crane
+		UncompressedTopLayerId: "placeholder",
+	}
+
+	// calling MarshalIndent so the json file written to disk is human-readable when opened
+	certImageJson, err := json.MarshalIndent(certImage, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	fileName, err := artifacts.WriteFile(viper.GetString("cert_image_file_name"), string(certImageJson))
```

Comment:
I am leaving a note here to keep track of checking for required values from viper (same as line 426)